### PR TITLE
Nix time

### PR DESCRIPTION
**Pull request by The1Penguin**

### CODE REVIEW

**Comment by The1Penguin** [comment]
This is a terrible idea.
It forces everyone to use nix to ensure that the builds will be reproducible.
The resulting image is around 40 times larger than it needs to be.
The action is probably wrong as well.
The positive part is that it will build faster.

**Review by Multipacker — CHANGES_REQUESTED** [review]
This is terrible

File: .dockerignore
Comment (malmz) [inline .dockerignore:1]: where it go??

File: .github/workflows/publish-docker-image.yml
Comment (malmz) [inline .github/workflows/publish-docker-image.yml:1]: i don't like this

File: Dockerfile
Comment (malmz) [inline Dockerfile:1]: docker? 🥺

File: docker-compose.yml
Comment (malmz) [inline docker-compose.yml:1]: docker compose deez nutz

File: flake.nix
Comment (malmz) [inline flake.nix:1]: don't get this, plz explan?

**Review by malmz — CHANGES_REQUESTED** [review]

File: .dockerignore
Comment (The1Penguin) [reply .dockerignore:1]: deleted

File: .github/workflows/publish-docker-image.yml
Comment (The1Penguin) [reply .github/workflows/publish-docker-image.yml:1]: Why not? :smiley:

File: Dockerfile
Comment (The1Penguin) [reply Dockerfile:1]: We got docker at home

File: flake.nix
Comment (The1Penguin) [reply flake.nix:1]: Okay, so ponder the following

What if your package manager, could build the entire docker image by itself, and handle the dependencies.
This does that, and a bit more ;P

**Review by Multipacker — COMMENTED** [review]
Much better

**Pull request closed without merging**